pyked/chemked.py

```diff
@@ -640,35 +640,109 @@ def __init__(self, properties):
 
+        Returns a formatted string no matter the type of composition. As such, this method
+        is not recommended for end users; instead, prefer the `get_cantera_mole_fraction`
+        or `get_cantera_mass_fraction` methods.
+
+        Arguments:
+            species_conversion (`dict`, optional): Mapping of species identifier to a
+                species name. This argument should be supplied when the name of the
+                species in the ChemKED YAML file does not match the name of the same
+                species in a chemical kinetic mechanism. The species identifier (the key
+                of the mapping) can be the name, InChI, or SMILES provided in the ChemKED
+                file, while the value associated with a key should be the desired name in
+                the Cantera format output string.
+
         Returns:
             `str`: String in the ``SPEC: AMT, SPEC: AMT`` format
+
+        Raises:
+            `ValueError`: If the composition type of the `DataPoint` is not one of
+                ``'mass fraction'``, ``'mole fraction'``, or ``'mole percent'``
         """
-        if self.composition_type == 'mole fraction':
-            return ', '.join(['{!s}:{:.4e}'.format(c['species-name'],
-                             c['amount'].magnitude) for c in self.composition]
-                             )
+        if self.composition_type in ['mole fraction', 'mass fraction']:
+            factor = 1.0
         elif self.composition_type == 'mole percent':
-            return ', '.join(['{!s}:{:.4e}'.format(c['species-name'],
-                             c['amount'].magnitude/100.0) for c in self.composition]
-                             )
+            factor = 100.0
         else:
-            raise ValueError('Cannot get mole fractions from the given composition.\n'
-                             '{}'.format(self.composition))
+            raise ValueError('Unknown composition type: {}'.format(self.composition_type))
 
-    def get_cantera_mass_fraction(self):
-        """Get the composition in a string format suitable for input to Cantera.
+        if species_conversion is None:
+            comps = ['{!s}:{:.4e}'.format(c['species-name'],
+                     c['amount'].magnitude/factor) for c in self.composition]
+        else:
+            comps = []
+            for c in self.composition:
+                amount = c['amount'].magnitude/factor
+                idents = [c.get(s) for s in ['species-name', 'InChI', 'SMILES'] if c.get(s, False)]
+                present = [i in species_conversion for i in idents]
+                if not any(present):
+                    comps.append('{!s}:{:.4e}'.format(c['species-name'], amount))
+                else:
+                    if len([i for i in present if i]) > 1:
+                        raise ValueError('More than one conversion present for species {}'.format(
+                                       c['species-name']))
```

Comment:
minor formatting issue, but can the second line align with the first, and have the final `)` on a new line, to match (what I think is) our style?